main.cpp

```diff
@@ -0,0 +1,225 @@
+    else if( r  >= 900 )
+        return int_to_roman( r - 900, rn + "CM" );
+    else if( r >= 500 )
+        return int_to_roman( r - 500, rn + "D" );
+    else if( r >= 400 )
+        return int_to_roman( r - 500, rn + "CD" );
+    else if( r >= 100 )
+        return int_to_roman( r - 100, rn + "C" );
+    else if( r >= 90 )
+        return int_to_roman( r - 90, rn + "XC" );
+    else if( r >= 50 )
+        return int_to_roman( r - 50, rn + "L" );
+    else if( r >= 40 )
+        return int_to_roman( r - 40, rn + "XL" );
+    else if( r >= 10 )
+        return int_to_roman( r - 10, rn + "X" );
+    else if( r >= 9 )
+        return int_to_roman( r - 9, rn + "IX" );
+    else if( r >= 5 )
+        return int_to_roman( r - 5, rn + "V" );
+    else if( r >= 4 )
+        return int_to_roman( r - 4, rn + "IV" );
+    else if ( r >= 1 )
+        return int_to_roman( r - 1, rn + "I" );
+    else
+        return rn;
+}
+
+string int_to_word( int w, string nw )
+{
+    const string ones[11] = {"","one","two","three","four","five","six","seven","eight","nine","ten"};
+    const string teens[10] = {"","eleven","twelve","thirteen","fourteen","fifteen",
+                             "sixteen","seventeen","eighteen","nineteen"};
+    const string tens[10] = {"","ten","twenty","thirty","fourty","fifty","sixty","seventy","eighty","ninety"};
+
+    string n;
+
+    if( ( w / 1000000 ) >= 1 )
+    {
+        n = ones[w/1000000];
+        w = w - ( (w/1000000) * 1000000 );
+        return int_to_word( w , nw + n + "million " );
+    }
+    else if( ( w / 100000 ) >= 1 )
+    {
+        n = ones[ w / 100000 ];
+        w = w - ( (w/100000) * 100000 );
+        return int_to_word( w , nw + n + "hundred-thousand " );
+    }
+    else if( ( w / 10000 ) > 1 )
+    {
+        n = tens[ w / 10000 ];
+        w = w - ( (w/10000) * 10000);
+        return int_to_word( w , nw + n + " " );
+    }
+    else if( ( w / 10000 ) == 1 )
+    {
+        n = teens[ (w - 10000) / 1000 ];
+        w = w - 10000;
+        w = w - ((w/1000) * 1000 );
+        return int_to_word( w , nw + n + " " + "thousand " );
+    }
+    else if( (w/1000) >= 1 )
+    {
+        n = ones[ w / 1000 ];
+        w = w - ( (w/1000) * 1000 );
+        return int_to_word( w , nw + n + " " + "thousand " );
+    }
+    else if( (w/100) >= 1 )
+    {
+        n = ones[ w / 100 ];
+        w = w - ( (w/100) * 100 );
+        return int_to_word( w , nw + n + " " + "hundred " );
+    }
+    else if( (w/10) > 1 )
+    {
+        n = tens[ w / 10 ];
+        w = w - ( (w/10) * 10 );
+        return int_to_word( w , nw + n );
+    }
+    else if( (w/10) == 1 )
+    {
+        n = teens[ w - 10 ];
+        return nw + n;
+    }
+    else
+    {
+        n = ones[ w / 1 ];
+        w = w - ( w/1 );
+        return nw + n;
+    }
+}
+
+int magic( int m )
+{
+    string _length = int_to_word( m );
+    int length_ = _length.length();
+    if( length_ == m )
+    {
+        cout << "4 is the magic number" << endl;
```

Comment:
just hard coding 4 like that? lol.  fun follow up question: is it always 4?  and, does this (or something similar) work in ever language? :-)